src/signals-unix.c

```diff
@@ -69,6 +69,43 @@ static void jl_exit_thread0(int signo, jl_bt_element_t *bt_data, size_t bt_size)
 int jl_simulate_longjmp(jl_jmp_buf mctx, bt_context_t *c) JL_NOTSAFEPOINT;
 static void jl_longjmp_in_ctx(int sig, void *_ctx, jl_jmp_buf jmpbuf);
 
+// create a fake function that describes the variable manipulations in jl_call_in_ctx/jl_call_in_state
+__attribute__((naked)) static void fake_stack_pop(void)
+{
+#if (defined(_OS_LINUX_) || defined(_OS_DARWIN_)) && defined(_CPU_X86_64_)
+    __asm__ volatile (
+        "  .cfi_signal_frame\n"
+        "  .cfi_def_cfa %rsp, 0\n" // CFA here uses %rsp directly
+        "  .cfi_offset %rip, 0\n" // previous value of %rip at CFA
+        "  .cfi_offset %rsp, 8\n" // previous value of %rsp at CFA
+        "  nop\n"
+    );
+#elif defined(_OS_LINUX_) && defined(_CPU_X86_)
+    __asm__ volatile (
+        "  .cfi_signal_frame\n"
+        "  .cfi_def_cfa %esp, 0\n" // CFA here uses %esp directly
+        "  .cfi_offset %eip, 0\n" // previous value of %eip at CFA
+        "  .cfi_offset %esp, 4\n" // previous value of %esp at CFA
+        "  nop\n"
+    );
+#elif (defined(_OS_LINUX_) || defined(_OS_DARWIN_)) && defined(_CPU_AARCH64_)
```

Comment:
```suggestion
#elif defined(_OS_DARWIN_) && defined(_CPU_AARCH64_)
```

GCC implements this function wrong (crashes the compiler)